src/Http/Http/src/ITagsMetadata.cs

```diff
@@ -0,0 +1,16 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Defines a contract used to specify a collection of tags in <see cref="Endpoint.Metadata"/>.
+    /// </summary>
+    public interface ITagsMetadata
+    {
+        /// <summary>
+        /// Gets the collection of tags associated with the endpoint.
+        /// </summary>
+        IList<string> Tags { get; }
```

Comment:
Should this be a readonly list?
```suggestion
        IReadOnlyList<string> Tags { get; }
```